src/mpyl/steps/deploy/dagster.py

```diff
@@ -28,6 +28,224 @@
+        )
+
+
+class TemplateDagster(Step):
+    """
+    This step creates a dagster user code helm chart manifest and writes an entry to the dagster server's configmap
+    but doesn't use helm to deloy the manifest.
+    """
+
+    def __init__(self, logger: Logger) -> None:
+        super().__init__(
+            logger,
+            Meta(
+                name="Dagster Template",
+                description="Creates a dagster user code helm chart and adds an entry to dagster's K8s ConfigMap",
+                version="0.0.1",
+                stage=STAGE_NAME,
+            ),
+            produced_artifact=ArtifactType.NONE,
+            required_artifact=ArtifactType.DOCKER_IMAGE,
+        )
+
+    def __evaluate_results(self, results: List[Output]):
+        return (
+            reduce(
+                self.__flatten_result_messages,
+                results[1:],
+                results[0],
+            )
+            if len(results) > 1
+            else results[0]
+        )
+
+    @staticmethod
+    def __flatten_result_messages(acc: Output, curr: Output) -> Output:
+        return Output(
+            success=acc.success and curr.success,
+            message=f"{acc.message}\n{curr.message}",
+        )
+
+    # pylint: disable=R0914
+    def execute(self, step_input: Input) -> Output:
+        """
+        Creates the dagster user-code helm chart manifest and adds an server entry to dagster server's ConfigMap
+        """
+        properties = step_input.run_properties
+        context = get_cluster_config_for_project(
+            step_input.run_properties, step_input.project_execution.project
+        ).context
+        dagster_config: DagsterConfig = DagsterConfig.from_dict(properties.config)
+        dagster_template_results = []
+
+        config.load_kube_config(context=context)
+        core_api = client.CoreV1Api()
+        apps_api = client.AppsV1Api()
+
+        name_suffix = get_name_suffix(properties)
+        release_name = convert_to_helm_release_name(
+            step_input.project_execution.name, name_suffix
+        )
+
+        builder = ChartBuilder(step_input)
+
+        user_code_deployment = to_user_code_values(
+            builder=builder,
+            release_name=release_name,
+            name_suffix=name_suffix,
+            run_properties=properties,
+            service_account_override=dagster_config.global_service_account_override,
+            docker_config=DockerConfig.from_dict(properties.config),
+        )
+
+        self._logger.debug(
+            f"Writing user code manifest with values: {user_code_deployment}"
+        )
+
+        values_path = Path(step_input.project_execution.project.target_path)
+        self._logger.info(f"Writing Helm values to {values_path}")
+
+        write_chart(
+            chart={},
+            chart_path=values_path,
+            chart_metadata="",
+            values=user_code_deployment,
+        )
+
+        if not step_input.dry_run:
```

Comment:
Everything above from this is a copy of the `Dagster Helm Template` step right? Maybe optimise that a bit to reduce the amount of copied code? Might also make it more clear which parts are shared

Reply:
Tried extracting big duplications to `DagsterBase`, wasnt too sure about the `rollout_restart` functions, but in theory, can do that too..

Reply:
[c788197](https://github.com/Vandebron/mpyl/pull/491/commits/c78819718b08d3dc12d6f4e5b77408dc5f06ed15)

heck, why not